File: SoccerPlayerApi/SoccerPlayerApi/Dtos/Facts/FactCreateDto.cs
using SoccerPlayerApi.Entities.Structure;
using System.Linq.Expressions;

namespace SoccerPlayerApi.Dtos.Facts;

public class FactCreateDto
{
    public decimal Amount { get; set; }

    public int DataTypeId { get; set; }

    public int Dimension1AggregationId { get; set; }
    public int? Dimension2AggregationId { get; set; }
    public int? Dimension3AggregationId { get; set; }
    public int? Dimension4AggregationId { get; set; }

    public int TimeAggregationId { get; set; }

    public int GetDimensionCount()
    {
        if (Dimension4AggregationId is not null && Dimension3AggregationId is not null && Dimension2AggregationId is not null) return 4;
        if (Dimension3AggregationId is not null && Dimension2AggregationId is not null) return 3;
        if (Dimension2AggregationId is not null) return 2;
        return 1;
    }

    public Expression<Func<Aggregation, bool>> ContainsAggregationId()
    {
        return aggregation => Dimension1AggregationId == aggregation.Id ||
                              Dimension2AggregationId == aggregation.Id ||
                              Dimension3AggregationId == aggregation.Id ||
                              Dimension4AggregationId == aggregation.Id;
    }
}

File: SoccerPlayerApi/clientapp/src/app/models/facts/factCreateDto.ts
export class FactCreateDto {
    Amount: number;
    DataTypeId: number;
    Dimension1AggregationId: number;
    Dimension2AggregationId?: number;
    Dimension3AggregationId?: number;
    Dimension4AggregationId?: number;
    TimeAggregationId: number;

    constructor(
        Amount: number,
        DataTypeId: number,
        TimeAggregationId: number,
        Dimension1AggregationId: number,
        Dimension2AggregationId?: number,
        Dimension3AggregationId?: number,
        Dimension4AggregationId?: number,
    ) {
        this.Amount = Amount;
        this.DataTypeId = DataTypeId;
        this.Dimension1AggregationId = Dimension1AggregationId;
        this.Dimension2AggregationId = Dimension2AggregationId;
        this.Dimension3AggregationId = Dimension3AggregationId;
        this.Dimension4AggregationId = Dimension4AggregationId;
        this.TimeAggregationId = TimeAggregationId;
    }
}
